elf/compat.go

```diff
@@ -0,0 +1,70 @@
+// +build linux
+
+// (c) 2018 Suchakra Sharma <suchakra@shiftleft.io>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package elf
+
+import (
+	"fmt"
+	"io/ioutil"
+	"regexp"
+	"runtime"
+)
+
+func GetSyscallFnName(name string) (string, error) {
+	var arch string
+	if runtime.GOARCH == "amd64" {
+		arch = "x64"
+	} else if runtime.GOARCH == "386" {
+		arch = "ia32"
+	} else {
+		// default
+		arch = "x64"
+	}
+
+	// We should search for new syscall function like "__x64__sys_open"
+	// Note the start of word boundary. Should return exactly one string
+	regexStr := `(\b__` + arch + `_[Ss]y[sS]_` + name + `\b)`
+	fnRegex := regexp.MustCompile(regexStr)
+
+	// Get kernel symbols
+	syms, err := ioutil.ReadFile("/proc/kallsyms")
+	if err != nil {
+		fmt.Println(string(syms))
+		return "", err
+	}
+
+	match := fnRegex.FindAllString(string(syms), -1)
+	// This should not be possible
+	if len(match) > 1 {
+		return "", err
```

Comment:
in this case `err` would be nil, should it be a useful message? maybe this check is not necessary if it's not a possible case

Reply:
Actually yes, we can remove this. It was just in case we get more than one match but thats not possible